tests/modules/modules_json.py

```diff
@@ -0,0 +1,214 @@
+import json
+import os
+import shutil
+
+from nf_core.modules.modules_json import ModulesJson
+from nf_core.modules.modules_repo import (
+    NF_CORE_MODULES_BASE_PATH,
+    NF_CORE_MODULES_NAME,
+    NF_CORE_MODULES_REMOTE,
+    ModulesRepo,
+)
+
+
+def test_get_modules_json(self):
+    """Checks that the get_modules_json function returns the correct result"""
+    mod_json_path = os.path.join(self.pipeline_dir, "modules.json")
+    with open(mod_json_path, "r") as fh:
+        mod_json_sb = json.load(fh)
+
+    mod_json_obj = ModulesJson(self.pipeline_dir)
+    mod_json = mod_json_obj.get_modules_json()
+
+    # Loop through all keys that should be present and check that
+    # they have the same values in the two dictionaries
+    keys = ["name", "homePage"]
+    for key in keys:
+        assert key in mod_json
+        assert mod_json[key] == mod_json_sb[key]
+
+    for repo in mod_json_sb["repos"]:
+        assert "base_path" in mod_json["repos"][repo]
+        assert "modules" in mod_json["repos"][repo]
+        for module in mod_json_sb["repos"][repo]["modules"]:
+            assert module in mod_json["repos"][repo]["modules"]
+            assert "git_sha" in mod_json["repos"][repo]["modules"][module]
+            assert (
+                mod_json["repos"][repo]["modules"][module]["git_sha"]
+                == mod_json_sb["repos"][repo]["modules"][module]["git_sha"]
+            )
```

Comment:
Are you looping through hardcoded keys to check that they are present?  Otherwise, wouldn't it be better to loop through all keys in `mod_json_sb` and check that the content is the same in `mod_json`?

Reply:
It also applies to test_mod_json_dump()